aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -332,14 +335,16 @@ __global__ void batch_norm_backward_kernel(
 
 // TensorAccessor in which the last dimensions are collapsed or expanded as needed
 template <typename scalar_t, int64_t dim>
-static PackedTensorAccessor<scalar_t, dim> reshaped_packed_accessor(const Tensor& t) {
+static PackedTensorAccessor<scalar_t, dim> reshaped_packed_accessor(const Tensor& t, bool forward) {
+  constexpr int too_small_feature_set = 16;  // this is the maximum feature dimension when we swap
+  constexpr int few_planes = 256;
   // undefined...
   if (! t.defined()) {
     const std::vector<int64_t> zeros(dim);
     return PackedTensorAccessor<scalar_t, dim>(nullptr, zeros.data(), zeros.data());
   }
   int64_t in_dim = t.dim();
-  if (in_dim == dim) {
+  if (in_dim == dim && (dim < 3 || (t.size(0) < t.size(2)) || (t.size(2) >= too_small_feature_set))) { // easy, if we don't need the evil trick
     return t.packed_accessor<scalar_t, dim>();
   }
 
```

Comment:
What is this `AT_CHECK` on line 351? Can't we instead call `.contiguous()` rather than erroring out?

Reply:
I would like to see some benchmark results on noncontig tensors, if this will be changed.